src/secure.ml

```diff
@@ -12,9 +12,11 @@ value base_dir_r = ref Filename.current_dir_name;
 value decompose =
   loop [] where rec loop r s =
     let b = Filename.basename s in
-    if b = "" || b = Filename.current_dir_name then
+    let cdn = Filename.current_dir_name in
+    let ds = Filename.dir_sep in
+    if b = "" || b = cdn || b = ds then
       let d = Filename.dirname s in
-      if d = "" || d = Filename.current_dir_name then r
+      if d = "" || d = cdn || d = ds then r
```

Comment:
Not sure the change on line 19 (|| d=ds) is really required to avoid infinite recursion.